hook!-full-game/addons/level_design_dock/plugin.gd

```diff
@@ -0,0 +1,111 @@
+tool
+extends EditorPlugin
+
+const INTERFACE_SCENE: = preload("res://addons/level_design_dock/LevelDesignDock.tscn")
+const LevelDesignDock: = preload("res://addons/level_design_dock/LevelDesignDock.gd")
+const PackedSceneButton: = preload("res://addons/level_design_dock/PackedSceneButton.gd")
+
+var _interface: LevelDesignDock
+var _current_instance: Node2D setget _set_current_instance
+var _current_parent: Node setget _set_current_parent
+var _current_scene: PackedScene setget _set_current_scene
+
+func _enter_tree() -> void:
+	_interface = INTERFACE_SCENE.instance()
+	add_control_to_dock(EditorPlugin.DOCK_SLOT_RIGHT_UR, _interface)
+	_interface.connect("packed_scene_button_created", self, "_on_Interface_PackedScene_Button_Created")
+
+
+func _exit_tree() -> void:
+	_interface.queue_free()
+
+
+func handles(object: Object) -> bool:
+	var can_handle = object is Node
+	if can_handle:
+		_set_current_parent(object)
+	return can_handle
+
+
+func forward_canvas_gui_input(event: InputEvent) -> bool:
+	var forward: = false
+	if event is InputEventMouseMotion:
+		_move_instance()
+		forward = true
+	elif event is InputEventMouseButton:
+		if event.button_index == BUTTON_LEFT:
+			forward = true
+			if event.is_pressed():
+				_create_instance()
+			else:
+				_clear_current_instance()
+	if event is InputEventKey:
+		if event.scancode == KEY_ESCAPE:
+			_clear_current_instance()
+			_clear_current_parent()
+			_clear_current_scene()
+			forward = true
+	return forward
+
+
+func _move_instance() -> void:
+	var can_drag = Input.is_mouse_button_pressed(BUTTON_LEFT) and _current_instance
+	if can_drag:
+		_current_instance.global_position = _current_instance.get_global_mouse_position()
+
+
+func _on_Interface_PackedScene_Button_Created(button: PackedSceneButton) -> void:
+	button.connect("toggled", self, "_on_PackedSceneButton_button_toggled", [button])
+
+
+func _on_PackedSceneButton_button_toggled(button_pressed: bool, button: PackedSceneButton):
+	if button_pressed:
+		_set_current_scene(button.packed_scene)
```

Comment:
The setter doesn't seem to do anything other than changing the property. In any case, the guideline is to use `self._current_scene = ...` to go through setters